### ISSUE TITLE
Difference between ContrastiveLoss and SupConLoss

### ISSUE THREAD

**Issue by ksasi** [KevinMusgrave/pytorch-metric-learning#483]
Hi,

I am trying to understand the difference between ContrastiveLoss and SupConLoss in the library. Aren't both the same in the sense both use labels to generate positive and negative pairs in a batch based on the labels? 

Please let me know if I am missing something in my understanding.

Thanks

**Comment by KevinMusgrave** [comment]
Yes they both use labels to generate positive and negative pairs. However the loss calculation differs. ContrastiveLoss applies a ReLU-like function to the pair distances, while SupConLoss uses something similar to softmax. So the gradients during backpropagation will be different.

**Comment by ksasi** [comment]
Thankyou for the explanation. Is there any StackOverflow or other place where we can ask questions/clarifications ?

**Comment by KevinMusgrave** [comment]
This is the best place to ask questions about this library
